File: src/main/java/team404/HardcorePlugin.java
package team404;

import org.bukkit.Bukkit;
import org.bukkit.plugin.java.JavaPlugin;
import team404.handlers.DevCommandExecutor;
import team404.handlers.InventoryListener;
import team404.handlers.PlayerRespawnListener;
import team404.models.Recipes;

import java.util.Optional;

import static team404.handlers.DevCommandExecutor.GET_RESOURCES_COMMAND;
import static team404.handlers.DevCommandExecutor.GET_STICK_COMMAND;

public final class HardcorePlugin extends JavaPlugin {
    private final PlayerRevivalService playerRevivalService = PlayerRevivalService.getInstance(this);

    @Override
    public void onEnable() {
        registerEvents();
        registerRecipes();
        registerCommands();
        createConfigFiles();
        playerRevivalService.loadData();
        MaterialLoader.loadMaterials(this.getDataFolder());
        getLogger().info("Hardcore plugin enabled");
    }

    private void createConfigFiles() {
        saveDefaultConfig();
        saveResource("materials/valuable_material_list.json", false);
        saveResource("materials/food_material_list.json", false);
        saveResource("materials/nature_material_list.json", false);
        saveResource("materials/ingredients_material_list.json", false);
    }
    private void registerEvents() {
        getServer().getPluginManager().registerEvents(new PlayerRespawnListener(this), this);
        getServer().getPluginManager().registerEvents(new InventoryListener(this), this);}
    private void registerRecipes() {
        Bukkit.addRecipe(Recipes.getReviveStuffRecipe(this));
    }
    private void registerCommands() {
        var commandExecutor = new DevCommandExecutor(this);
        Optional.ofNullable(getCommand(GET_RESOURCES_COMMAND))
                .ifPresent(it -> it.setExecutor(commandExecutor));
        Optional.ofNullable(getCommand(GET_STICK_COMMAND))
                .ifPresent(it -> it.setExecutor(commandExecutor));
    }
}

File: src/main/kotlin/team404/HardcorePlugin.kt
package team404

import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin
import team404.listener.InventoryListener
import team404.listener.PlayerRespawnListener
import team404.listener.command.BuybackCommandExecutor
import team404.listener.command.BuybackCommandExecutor.Companion.BUYBACK_COMMAND
import team404.listener.command.DevCommandExecutor
import team404.listener.command.DevCommandExecutor.Companion.GET_RESOURCES_COMMAND
import team404.listener.command.StickCommandExecutor
import team404.listener.command.StickCommandExecutor.Companion.GET_STICK_COMMAND
import team404.model.Recipes
import team404.service.MaterialLoader
import team404.service.PlayerRevivalService

class HardcorePlugin : JavaPlugin() {
    private val playerRevivalService = PlayerRevivalService.getInstance(this)

    override fun onEnable() {
        registerEvents()
        registerRecipes()
        registerCommands()
        createConfigFiles()
        playerRevivalService.loadData()
        MaterialLoader.loadMaterials(this.dataFolder)
        logger.info("Hardcore plugin enabled")
    }

    private fun createConfigFiles() {
        saveDefaultConfig()
        saveResource("materials/valuable_material_list.json", false)
        saveResource("materials/food_material_list.json", false)
        saveResource("materials/nature_material_list.json", false)
        saveResource("materials/ingredients_material_list.json", false)
    }

    private fun registerEvents() {
        server.pluginManager.registerEvents(PlayerRespawnListener(this), this)
        server.pluginManager.registerEvents(InventoryListener(this), this)
    }

    private fun registerRecipes() {
        Bukkit.addRecipe(Recipes.getReviveStuffRecipe())
    }

    private fun registerCommands() {
        getCommand(GET_RESOURCES_COMMAND)?.setExecutor(DevCommandExecutor(this))
        getCommand(GET_STICK_COMMAND)?.setExecutor(StickCommandExecutor())
        getCommand(BUYBACK_COMMAND)?.setExecutor(BuybackCommandExecutor(this))
    }
}
